frontman_test.go

```diff
@@ -12,3 +13,33 @@ func helperCreateFrontman(t *testing.T, cfg *Config) *Frontman {
 	assert.Nil(t, err)
 	return fm
 }
+
+func TestFrontmanHubInput(t *testing.T) {
+	hub := NewMockHub("localhost:9100")
+	go hub.Serve()
+
+	cfg, err := HandleAllConfigSetup(DefaultCfgPath)
+	assert.Nil(t, err)
+
+	cfg.HubURL = hub.URL() + "/?serviceChecks=10&webChecks=10"
+	cfg.LogLevel = "debug"
+	cfg.Sleep = 1           // delay between each round of checks
+	cfg.SenderBatchSize = 2 // number of results to send to hub at once
+	cfg.SenderInterval = 1
+	cfg.ICMPTimeout = 0.1
+	cfg.HTTPCheckTimeout = 1.0
+
+	fm := helperCreateFrontman(t, cfg)
+
+	resultsChan := make(chan Result, 100)
+	interruptChan := make(chan struct{})
+
+	go fm.Run("", nil, interruptChan, resultsChan)
+
+	// stop after some time
+	time.Sleep(5 * time.Second)
```

Comment:
is there a smart way to identify that the go routine has finished it's job as waiting for 6 seconds would make all tests execution very slow

Reply:
Since this test involves running random webchecks, we can't determine in advance the time it will take to finish, so we use a 5 second deadline in order to guarantee some results.

Reply:
what I mean is that the background tests report via channel if they are ready so you can finish main routine before 5 seconds if the tests are finished. on the other hand if they need more than 5 seconds it doesn't mean they are failing right? 
`	select {
		case <-ready: //tests are ready
			return nil
		case <-time.After(time.Second * 20): //timout
			return fmt.Errorf("timeout")
	}`
You can also avoid timeouts if you pass to go routines a timeout context so the calls will be skipped after the give time interval.

Reply:
It's a good idea! I'll look into fixing this right now. I also want to avoid slow tests in any way possible. We just had no test covering this code at all before this so anything was better than before.

Reply:
Reworked test in a less intrusive way for now, from 6 sec to 0.2 sec
```
--- PASS: TestFrontmanHubInput (0.20s)
```